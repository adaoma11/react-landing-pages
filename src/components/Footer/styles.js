import styled, { css } from 'styled-components';

export const Footer = styled.footer`
  ${({ theme, $hasBg }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    height: 150px;
    background-color: ${$hasBg ? theme.colors.primary : theme.colors.white};
  `}
`;
