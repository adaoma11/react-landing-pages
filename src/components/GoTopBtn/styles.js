import styled, { css } from 'styled-components';

export const Link = styled.a`
  ${({ theme }) => css`
    display: flex;
    width: 40px;
    height: 40px;
    position: fixed;
    bottom: 2rem;
    right: 2rem;
    color: ${theme.colors.white};
    background-color: ${theme.colors.primary};
    border-radius: 5px;
    opacity: 0.8;
  `}
`;
