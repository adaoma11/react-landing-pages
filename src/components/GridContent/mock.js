export default {
  title: 'Grid Component Example',
  text: `Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptate minus repellendus fugiat aliquam neque corporis! Quidem, deserunt! Rem quod quo amet doloremque, excepturi doloribus, provident nesciunt blanditiis consequuntur facere quia!`,
  textAlign: 'justify',
  hasBg: false,
  titleUpperCase: true,
};
