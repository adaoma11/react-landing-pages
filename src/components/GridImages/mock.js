export default {
  title: 'Grid Images Example',
  description: 'Uma breve descrição sobre este album',
  gridItems: [
    {
      altText: 'Imagem 1',
      imgSrc: 'https://picsum.photos/300?random=1',
    },
    {
      altText: 'Imagem 2',
      imgSrc: 'https://picsum.photos/300?random=2',
    },
    {
      altText: 'Imagem 3',
      imgSrc: 'https://picsum.photos/300?random=3',
    },
    {
      altText: 'Imagem 4',
      imgSrc: 'https://picsum.photos/300?random=4',
    },
    {
      altText: 'Imagem 5',
      imgSrc: 'https://picsum.photos/300?random=5',
    },
    {
      altText: 'Imagem 6',
      imgSrc: 'https://picsum.photos/300?random=6',
    },
  ],
  hasBg: false,
  titleUpperCase: false,
};
