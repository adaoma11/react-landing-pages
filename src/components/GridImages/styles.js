import styled from 'styled-components';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  width: 100%;
  gap: 20px;
  margin-top: 50px;
`;

export const GridItem = styled.div`
  overflow: hidden;
`;

export const Image = styled.img`
  width: 100%;
  transition: all 0.5s ease-in-out;

  &:hover {
    transform: scale(1.2) rotate(5deg);
  }
`;
