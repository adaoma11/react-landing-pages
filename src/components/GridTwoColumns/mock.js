export default {
  title: 'Grid Two Columns',
  text: `Lorem ipsum dolor, sit amet consectetur adipisicing elit. Quia iusto omnis magnam libero ea neque deserunt optio aliquam, quas esse fugit nulla, quaerat tempora repellat? Cupiditate voluptatum sint suscipit facilis.`,
  textAlign: 'justify',
  imgSrc: 'vite.svg',
  hasBg: false,
  titleUpperCase: true,
};
