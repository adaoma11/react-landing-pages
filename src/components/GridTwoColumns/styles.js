import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    display: grid;
    grid-template-columns: 1fr 1fr;
    align-items: center;
    gap: 30px;

    @media ${theme.medias.medium} {
      grid-template-columns: 1fr;
    }
  `}
`;

export const TextContainer = styled.div`
  display: flex;
  flex-direction: column;
  gap: 30px;
`;

export const Image = styled.div`
  ${({ $imgSrc }) => css`
    display: block;
    min-height: 500px;
    background-image: url(${$imgSrc});
    background-repeat: no-repeat;
    background-position: center;
    background-size: 60%;
  `}
`;
