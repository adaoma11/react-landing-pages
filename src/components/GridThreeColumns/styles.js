import styled from 'styled-components';
import { Title as GridItemTitle } from '../Heading/styles';

export const GridContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
  counter-reset: grid-counter;
  width: 100%;
  gap: 50px;
`;

export const GridItem = styled.div`
  margin-top: 80px;

  ${GridItemTitle} {
    position: relative;
    left: 4rem;
    margin-bottom: 30px;
  }

  ${GridItemTitle}::before {
    counter-increment: grid-counter;
    content: counter(grid-counter);
    position: absolute;
    font-size: 5rem;
    top: -2rem;
    left: -4rem;
    transform: rotate(7deg);
  }
`;
