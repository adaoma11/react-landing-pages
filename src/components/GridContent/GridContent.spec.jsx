import { GridContent } from '.';
import mock from './mock';
import { renderTheme } from '../../styles/renderTheme';

describe('<GridContent />', () => {
  it('should match snapshot', () => {
    const { container } = renderTheme(<GridContent {...mock} />);
    expect(container).toMatchSnapshot();
  });
});
