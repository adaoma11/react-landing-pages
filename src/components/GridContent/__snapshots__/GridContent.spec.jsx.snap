// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridContent /> should match snapshot 1`] = `
.c2 {
  display: flex;
  flex-direction: column;
  gap: 30px;
}

.c0 {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  min-height: 100vh;
  background-color: #FFFFFF;
}

.c1 {
  max-width: 1200px;
  padding: 30px;
}

.c3 {
  color: #0A1128;
  font-size: 3rem;
  text-transform: uppercase;
}

.c4 {
  font-family: 'Poppins',sans-serif;
  font-size: 1.5rem;
  color: #0A1128;
  text-align: justify;
  font-weight: normal;
}

@media (max-width: 768px) {
  .c3 {
    font-size: 2rem;
  }
}

<div>
  <section
    class="c0"
  >
    <article
      class="c1"
    >
      <div
        class="c2"
      >
        <h2
          class="c3"
        >
          Grid Component Example
        </h2>
        <p
          class="c4"
        >
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptate minus repellendus fugiat aliquam neque corporis! Quidem, deserunt! Rem quod quo amet doloremque, excepturi doloribus, provident nesciunt blanditiis consequuntur facere quia!
        </p>
      </div>
    </article>
  </section>
</div>
`;
